airflow/providers/apache/kafka/hooks/kafka_producer.py

```diff
@@ -0,0 +1,88 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from kafka import KafkaProducer
+
+from airflow.hooks.base_hook import BaseHook
+
+
+class KafkaProducerHook(BaseHook):
+    """KafkaProducerHook Class."""
+
+    DEFAULT_HOST = 'localhost'
+    DEFAULT_PORT = 9092
+
+    def __init__(self, conn_id, topic):
+        super().__init__(None)
+        self.conn_id = conn_id
+        self._conn = None
+        self.server = None
+        self.consumer = None
+        self.producer = None
+        self.topic = topic
+
+    def get_conn(self) -> KafkaProducer:
+        """
+            Returns a Kafka Producer
+
+        :return:
+            A Kafka Producer object.
+        """
+        if not self._conn:
```

Comment:
This attribute is never set. In other words, this condition is always met. Is it expected?

Reply:
The `_conn` attribute is set on line 47 using the base_hook `get_connection` method.  This expression will be true once during the first call to `get_conn` where it is initialized on lines 47 - 53.  Thereafter, it returns the KafkaProducer object.

On that note, I've changed line 53 to say `self.producer =` and removed the `consumer` attribute entirely